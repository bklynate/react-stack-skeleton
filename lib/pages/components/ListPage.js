'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ListPage;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

var _list = require('../../list');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * A page that display the todo list.
 */
function ListPage(_ref) {
  var otherProps = _objectWithoutProperties(_ref, []);

  return _react2.default.createElement(
    'section',
    { className: 'todoapp' },
    _react2.default.createElement(_list.ListHeader, null),
    _react2.default.createElement(
      _list.ListBody,
      null,
      _react2.default.createElement(_list.ListItem, { itemId: 0 }),
      _react2.default.createElement(_list.ListItem, { itemId: 1 }),
      _react2.default.createElement(_list.ListItem, { itemId: 2 })
    ),
    _react2.default.createElement(_list.ListFooter, null)
  );
}