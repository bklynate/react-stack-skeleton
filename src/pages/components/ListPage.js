/* @flow */
import React from 'react';
import { Link } from 'react-router';

import { ListBody, ListFooter, ListHeader, ListItem } from '../../list';

type Props = {

};

/**
 * A page that display the todo list.
 */
export default function ListPage({ ...otherProps }: Props): React.Element<*> {
  return (
    <section className="todoapp">
      <ListHeader />
      <ListBody>
        <ListItem itemId={0} />
        <ListItem itemId={1} />
        <ListItem itemId={2} />
      </ListBody>
      <ListFooter />
    </section>
  );
}
