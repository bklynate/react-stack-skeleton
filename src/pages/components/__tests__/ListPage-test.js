import React from 'react';
import renderer from 'react-test-renderer';

import ListPage from '../ListPage';

test('it displays the "NYI" message.', () => {
  const component = renderer.create(
    <ListPage />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
