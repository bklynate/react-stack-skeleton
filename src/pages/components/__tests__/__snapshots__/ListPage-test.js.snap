exports[`test it displays the "NYI" message. 1`] = `
<section
  className="todoapp">
  <header
    className="header">
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onChange={[Function]}
      placeholder="What needs to be done?"
      value="" />
  </header>
  <section
    className="main">
    <input
      className="toggle-all"
      onChange={[Function]}
      type="checkbox" />
    <label
      htmlFor="toggle-all">
      Mark all as complete
    </label>
    <ul
      className="todo-list">
      <li
        className={undefined}>
        <div
          className="view">
          <input
            checked={false}
            className="toggle"
            onChange={[Function]}
            type="checkbox" />
          <label>
            Test label 0
          </label>
          <button
            className="destroy" />
        </div>
        <input
          className="edit"
          onChange={[Function]}
          value="Test value 0" />
      </li>
      <li
        className={undefined}>
        <div
          className="view">
          <input
            checked={false}
            className="toggle"
            onChange={[Function]}
            type="checkbox" />
          <label>
            Test label 1
          </label>
          <button
            className="destroy" />
        </div>
        <input
          className="edit"
          onChange={[Function]}
          value="Test value 1" />
      </li>
      <li
        className="completed">
        <div
          className="view">
          <input
            checked={true}
            className="toggle"
            onChange={[Function]}
            type="checkbox" />
          <label>
            Test label 2
          </label>
          <button
            className="destroy" />
        </div>
        <input
          className="edit"
          onChange={[Function]}
          value="Test value 2" />
      </li>
    </ul>
  </section>
  <footer
    className="footer">
    <span
      className="todo-count">
      <strong>
        2
      </strong>
       item left
    </span>
    <ul
      className="filters">
      <li>
        <a
          className="selected"
          href="/"
          onClick={[Function]}
          style={Object {}}>
          All
        </a>
      </li>
      <li>
        <a
          className={undefined}
          href="/active"
          onClick={[Function]}
          style={Object {}}>
          Active
        </a>
      </li>
      <li>
        <a
          className={undefined}
          href="/completed"
          onClick={[Function]}
          style={Object {}}>
          Completed
        </a>
      </li>
    </ul>
    <button
      className="clear-completed"
      onClick={[Function]}>
      Clear completed
    </button>
  </footer>
</section>
`;
